### ISSUE TITLE
Instructions did not work for OSX

### ISSUE THREAD

**Issue by gauravsaxena81** [morgant/finishGoogleCodeGitHubWikiMigration#2]
The instructions did not work for OSX. Got following error

```
Cloning 'gauravsaxena81/bit-dekk'... Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error!
Exiting.
```

Finally what worked for me was I turned https=true in the script and used url of the form 

```
git clone https://github.com/YOUR_USERNAME/YOUR_REPOSITORY
```

This may need to be added to the instructions.

**Comment by morgant** [comment]
I'm sorry it didn't go smoothly for you, but thanks for the feedback. I've got a couple questions that might help me resolve this issue for other users in the future:
1. In your first attempt, I assume that you used an SSH clone URL (e.g. `git@github.com:USERNAME/REPOSITORY`), correct? If so, had you already set up [GitHub SSH keys](https://help.github.com/articles/generating-ssh-keys/)?
2. In your final solution to use the HTTPS URL for the repository, did it not work with just the URL and you were forced to modify the script to change it to `https=true`?

**Comment by gauravsaxena81** [comment]
Thanks for the tool. It really made my life simple.

To answer your questions:
1. Yes I did use ssh clone url. I did not set up ssh keys. Was it part of
   the instructions? I may have missed it.
2. Cursory reading of script revealed that if I turn http=true, a https URL
   of the repo is expected which has a different structure( https://github.com/
   _YOUR_USERNAME_/_YOUR_REPOSITORY)_. So I turned it to true. I did not try
   running the script with https=false

Hope it helps. Do let me know if you need more information.

On Thursday, July 23, 2015, Morgan Aldridge notifications@github.com
wrote:

> I'm sorry it didn't go smoothly for you, but thanks for the feedback. I've
> got a couple questions that might help me resolve this issue for other
> users in the future:
> 1. In your first attempt, I assume that you used an SSH clone URL
>    (e.g. git@github.com:USERNAME/REPOSITORY), correct? If so, had you
>    already set up GitHub SSH keys
>    https://help.github.com/articles/generating-ssh-keys/?
> 2. In your final solution to use the HTTPS URL for the repository, did
>    it not work with just the URL and you were forced to modify the script to
>    change it to https=true?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/morgant/finishGoogleCodeGitHubWikiMigration/issues/2#issuecomment-124154469
> .

## 

Regards
Gaurav Saxena

**Comment by dorkbox** [comment]
Also, if using HTTPS, the repo name must be lowercase, otherwise the regex won't match on it.

This fixes it: https?://github.com/([-a-zA-Z0-9]+)/([-a-zA-Z0-9]+))

**Comment by dorkbox** [comment]
This fixes it: https?://github.com/([-a-zA-Z0-9]+)/([-a-zA-Z0-9]+))

**Comment by morgant** [comment]
My apologies for my delay in updating this tool and responding.

@gauravsaxena81 It looks like the regex for the HTTPS clone support was broken and was able to clone the main repository, but not the wiki repository. I've fixed this in the latest release. I also updated the README to note the requirement of setting up SSH keys (if using SSH clone URLs).

@dorkbox Thanks for pointing out the case-sensitivity issue with the regex. I have also fixed this in the latest release.
